src/Api/AbstractApi.php

```diff
@@ -7,57 +7,14 @@
 use Psr\Http\Message\ResponseInterface as HttpResponse;
 use Silktide\ProspectClient\Data\BodyData;
 use Silktide\ProspectClient\Data\QueryStringData;
+use Silktide\ProspectClient\HTTP\HttpWrapper;
 
 abstract class AbstractApi
 {
-    const API_SCHEME = "https";
-    const API_HOST = "api.prospect.silktide.com";
-    const API_PATH_VERSION = "/api/v1";
+    protected $httpWrapper;
 
-    private string $apiKey;
-    private HttpClient $httpClient;
-
-    public function __construct(string $apiKey, HttpClient $httpClient = null)
+    public function __construct(HttpWrapper $httpWrapper)
```

Comment:
By moving the logic to an HttpWrapper, we abstract away the Api's having to know anything about the apiKey.

It also reduces the boilerplate in the Abstract class and potentially allows the `callApi` functionality to be tested independently